chomp-frontend/src/side-bar/SideBar.tsx

```diff
@@ -0,0 +1,163 @@
+import * as React from 'react';
+import { Link } from 'react-router-dom'; // Ensure this is from 'react-router-dom' to handle routing
+import { useState } from 'react';
+import { useEffect } from 'react';
+
+import johnImage from '../assets/images/john.png';
+import janeImage from '../assets/images/jane.png';
+import aliceImage from '../assets/images/alice.jpg';
+import bobImage from '../assets/images/bob.jpg';
+
+import sidebarEdgeImage from "../assets/images/sidebar-edge-2.png";
+
+const dummyUsers = [
+    { id: 1, name: "John Doe", profilePicture: johnImage },
+    { id: 2, name: "Jane Smith", profilePicture: janeImage },
+    { id: 3, name: "Alice Johnson", profilePicture: aliceImage },
+    { id: 4, name: "Bob Brown", profilePicture: bobImage },
+];
+
+const dummyCommunities = [
+    { id: 1, name: "TapasConsumers", profilePicture: johnImage },
+    { id: 2, name: "Easy Breakfast", profilePicture: janeImage },
+    { id: 3, name: "Weekly Planning", profilePicture: aliceImage },
+    { id: 4, name: "Vegan Prepping", profilePicture: bobImage },
+]
+
+export function SideBar() {
+    const [isOpen, setIsOpen] = useState(true);
+    const toggleSidebar = () => setIsOpen(!isOpen);
+
+    useEffect(() => {
+        const handleResize = () => {
+          if (window.innerWidth < 768) {
+            setIsOpen(false);
+          } else {
+            setIsOpen(true);
+          }
+        };
+      
+        handleResize(); // check on mount
+        window.addEventListener('resize', handleResize);
+        return () => window.removeEventListener('resize', handleResize);
+      }, []);
+
+    return (
+        <div
+            id="side-bar"
+            className={`h-screen ${
```

Comment:
```suggestion
            className={`h-auto ${
```
when testing the page it causes rendering issues when the screen is too short vertically, h-auto appears to fix this on my end